services/management/management.go

```diff
@@ -207,33 +217,38 @@ func (s *service) initMetrics(metricFactory metric.Factory) {
 func (s *service) updateMetrics() {
 	s.RLock()
 	defer s.RUnlock()
-	s.metrics.currentRefTime.Update(int64(s.data.CurrentReference))
+
+	currentRef := s.data.CurrentReference
+
+	s.metrics.currentRefTime.Update(int64(currentRef))
 	s.metrics.genesisRefTime.Update(int64(s.data.GenesisReference))
 	s.metrics.lasUpdateTime.Update(time.Now().Unix())
-	s.metrics.currentTopology.Update(fmt.Sprintf("%v", s.data.CurrentTopology))
+	topologyStringArray := make([]string, 0, len(s.data.CurrentTopology))
+	for _, peer := range s.data.CurrentTopology {
+		topologyStringArray = append(topologyStringArray, fmt.Sprintf("{\"address:\":\"%s\", \"ip\":\"%s\", \"port\":%d}", peer.HexOrbsAddress(), peer.GossipEndpoint(), peer.GossipPort()))
```

Comment:
1) you need to have json marshal to the primitives.NodeAddress which we don't
2) i don't need the map (the key is a keyForMap() unreadable)
3) since nodeaddress has an override to string it can cause problems with marshal